modyn/storage/src/internal/file_wrapper/binary_file_wrapper.cpp

```diff
@@ -61,13 +61,24 @@ std::vector<int64_t> BinaryFileWrapper::get_all_labels() {
   std::vector<int64_t> labels = std::vector<int64_t>();
   labels.reserve(num_samples);
 
-  for (uint64_t i = 0; i < num_samples; ++i) {
-    get_stream()->seekg(static_cast<int64_t>(i * record_size_), std::ios::beg);
-
-    std::vector<unsigned char> label_vec(label_size_);
-    get_stream()->read(reinterpret_cast<char*>(label_vec.data()), static_cast<int64_t>(label_size_));
-
-    labels.push_back(int_from_bytes(label_vec.data(), label_vec.data() + label_size_));
+  std::vector<unsigned char> tmp_label_buffer(label_size_);
+
+  // Duplicating the loop here to avoid doing the endianess check in every loop iteration;
+  // there might be more idomatic ways to do this though.
```

Comment:
We should use a function pointer here instead. Something like

```
auto int_from_bytes = little_endian_ ? int_from_bytes_little_endian : int_from_bytes_big_endian
``` 

and then just call `int_from_bytes`. No need to duplicate the loop, then :)

Reply:
That's neat, wasn't sure if this works in c++ :)